base/applications/regedit/framewnd.c

```diff
@@ -807,6 +814,98 @@ static void ChooseFavorite(LPCWSTR pszFavorite)
         RegCloseKey(hKey);
 }
 
+static LPWSTR GetItemFullPath(HTREEITEM hTI)
+{
+    HKEY hRoot;
+    WCHAR rootname[MAX_PATH], *buffer;
+    SIZE_T rootlen, subkeylen;
+    LPCWSTR subkey = GetItemPath(g_pChildWnd->hTreeWnd, hTI, &hRoot);
+    if (!subkey || !hRoot)
+        return NULL;
+    if (!GetKeyName(rootname, ARRAY_SIZE(rootname), hRoot, L""))
+        return NULL;
+    rootlen = lstrlenW(rootname) + 1; // + 1 for '\\'
+    subkeylen = lstrlenW(subkey);
+    buffer = (WCHAR*)malloc((rootlen + subkeylen + 1) * sizeof(WCHAR));
+    if (buffer)
+    {
+        lstrcpyW(buffer, rootname);
```

Comment:
We are already using <strsafe.h>.
Please use `StringCchCopyW` function.

Reply:
> for code safety

I can guarantee you that "HKEY_LOCAL_MACHINE" is going to fit in a MAX_PATH buffer.

Reply:
This codebase is for regular human reading.
The god robots don't trust human logical sence.

Reply:
```suggestion
        StringCchCopyW(buffer, _countof(buffer), rootname);
```

Reply:
> _countof(buffer)

No, that is broken code!

Reply:
Sorry.